persepolisdm/persepolis
# New persepolis python package!

## Notification sounds is removed!
From now persepolis is using freedesktop notification sounds in Gnu\Linux and BSD. 
* In GNU/Linux and BSD sound-theme-freedesktop package must be installed.

## Installation method is changed!
* python3-setuptools (for debian base distro*) and python-setuptools for arch must be installed

## How to install:

1. Clone project.
2. go to the project folder whit cd command.
3. you  can install persepolis by this command(this script only works in Gnu\Linux and BSD for now).
```
sudo python setup.py install
```
## How to uninstall :
 you can uninstall persepolis by : 

```
sudo python uninstall.py
```
## How to test:
for testing persepolis with out installation:
1. make sure dependencies is installed.
2. navigate in test folder with cd command.
3. run :
```
python test.py
```
## new tree of project: 
```
.
├── icons
│   ├── Archdroid-Blue
│   │   ├── about.svg
│   │   ├── add_queue.svg
│   │   ├── add.svg
│   │   ├── browser.svg
│   │   ├── down.svg
│   │   ├── exit.svg
│   │   ├── file.svg
│   │   ├── folder.svg
│   │   ├── menu.svg
│   │   ├── minimize.svg
│   │   ├── multi_down.svg
│   │   ├── multi_remove.svg
│   │   ├── multi_trash.svg
│   │   ├── multi_up.svg
│   │   ├── ok.svg
│   │   ├── pause_queue.svg
│   │   ├── pause.svg
│   │   ├── play.svg
│   │   ├── preferences.svg
│   │   ├── remove_queue.svg
│   │   ├── remove.svg
│   │   ├── select_all.svg
│   │   ├── setting.svg
│   │   ├── start_queue.svg
│   │   ├── stop_all.svg
│   │   ├── stop_queue.svg
│   │   ├── stop.svg
│   │   ├── trash.svg
│   │   ├── up.svg
│   │   └── window.svg
│   ├── Archdroid-Red
│   │   ├── about.svg
│   │   ├── add_queue.svg
│   │   ├── add.svg
│   │   ├── browser.svg
│   │   ├── down.svg
│   │   ├── exit.svg
│   │   ├── file.svg
│   │   ├── folder.svg
│   │   ├── menu.svg
│   │   ├── minimize.svg
│   │   ├── multi_down.svg
│   │   ├── multi_remove.svg
│   │   ├── multi_trash.svg
│   │   ├── multi_up.svg
│   │   ├── ok.svg
│   │   ├── pause_queue.svg
│   │   ├── pause.svg
│   │   ├── play.svg
│   │   ├── preferences.svg
│   │   ├── remove_queue.svg
│   │   ├── remove.svg
│   │   ├── select_all.svg
│   │   ├── setting.svg
│   │   ├── start_queue.svg
│   │   ├── stop_all.svg
│   │   ├── stop_queue.svg
│   │   ├── stop.svg
│   │   ├── trash.svg
│   │   ├── up.svg
│   │   └── window.svg
│   ├── Breeze
│   │   ├── about.svg
│   │   ├── add_queue.svg
│   │   ├── add.svg
│   │   ├── browser.svg
│   │   ├── down.svg
│   │   ├── exit.svg
│   │   ├── file.svg
│   │   ├── folder.svg
│   │   ├── menu.svg
│   │   ├── minimize.svg
│   │   ├── multi_down.svg
│   │   ├── multi_remove.svg
│   │   ├── multi_trash.svg
│   │   ├── multi_up.svg
│   │   ├── ok.svg
│   │   ├── pause_queue.svg
│   │   ├── pause.svg
│   │   ├── play.svg
│   │   ├── preferences.svg
│   │   ├── remove_queue.svg
│   │   ├── remove.svg
│   │   ├── select_all.svg
│   │   ├── setting.svg
│   │   ├── start_queue.svg
│   │   ├── stop_all.svg
│   │   ├── stop_queue.svg
│   │   ├── stop.svg
│   │   ├── trash.svg
│   │   ├── up.svg
│   │   └── window.svg
│   ├── Breeze-Dark
│   │   ├── about.svg
│   │   ├── add_queue.svg
│   │   ├── add.svg
│   │   ├── browser.svg
│   │   ├── down.svg
│   │   ├── exit.svg
│   │   ├── file.svg
│   │   ├── folder.svg
│   │   ├── menu.svg
│   │   ├── minimize.svg
│   │   ├── multi_down.svg
│   │   ├── multi_remove.svg
│   │   ├── multi_trash.svg
│   │   ├── multi_up.svg
│   │   ├── ok.svg
│   │   ├── pause_queue.svg
│   │   ├── pause.svg
│   │   ├── play.svg
│   │   ├── preferences.svg
│   │   ├── remove_queue.svg
│   │   ├── remove.svg
│   │   ├── select_all.svg
│   │   ├── setting.svg
│   │   ├── start_queue.svg
│   │   ├── stop_all.svg
│   │   ├── stop_queue.svg
│   │   ├── stop.svg
│   │   ├── trash.svg
│   │   ├── up.svg
│   │   └── window.svg
│   ├── icons_resource.qrc
│   ├── icons_resources_generator.sh
│   ├── persepolis.png
│   └── persepolis.svg
├── LICENSE
├── man
│   ├── persepolis.1
├── persepolis
│   ├── gui
│   │   ├── about_ui.py
│   │   ├── addlink_ui.py
│   │   ├── after_download_ui.py
│   │   ├── icons_resource.py
│   │   ├── __init__.py
│   │   ├── mainwindow_ui.py
│   │   ├── palettes.py
│   │   ├── progress_ui.py
│   │   │   ├── about_ui.cpython-36.pyc
│   │   │   ├── addlink_ui.cpython-36.pyc
│   │   │   ├── after_download_ui.cpython-36.pyc
│   │   │   ├── icons_resource.cpython-36.pyc
│   │   │   ├── __init__.cpython-36.pyc
│   │   │   ├── mainwindow_ui.cpython-36.pyc
│   │   │   ├── palettes.cpython-36.pyc
│   │   │   ├── progress_ui.cpython-36.pyc
│   │   │   ├── setting_ui.cpython-36.pyc
│   │   │   └── text_queue_ui.cpython-36.pyc
│   │   ├── setting_ui.py
│   │   └── text_queue_ui.py
│   ├── __init__.py
│   ├── __main__.py
│   │   ├── __init__.cpython-36.pyc
│   │   └── __main__.cpython-36.pyc
│   └── scripts
│       ├── about.py
│       ├── addlink.py
│       ├── after_download.py
│       ├── browser_integration.py
│       ├── bubble.py
│       ├── chromium_integration_window.py
│       ├── compatibility.py
│       ├── download.py
│       ├── error_window.py
│       ├── flashgot_queue.py
│       ├── initialization.py
│       ├── __init__.py
│       ├── logger.py
│       ├── mac_notification.py
│       ├── mainwindow.py
│       ├── newopen.py
│       ├── osCommands.py
│       ├── persepolis.py
│       ├── play.py
│       ├── progress.py
│       ├── properties.py
│       │   ├── about.cpython-36.pyc
│       │   ├── addlink.cpython-36.pyc
│       │   ├── after_download.cpython-36.pyc
│       │   ├── browser_integration.cpython-36.pyc
│       │   ├── bubble.cpython-36.pyc
│       │   ├── chromium_integration_window.cpython-36.pyc
│       │   ├── compatibility.cpython-36.pyc
│       │   ├── download.cpython-36.pyc
│       │   ├── error_window.cpython-36.pyc
│       │   ├── flashgot_queue.cpython-36.pyc
│       │   ├── __init__.cpython-36.pyc
│       │   ├── initialization.cpython-36.pyc
│       │   ├── logger.cpython-36.pyc
│       │   ├── mainwindow.cpython-36.pyc
│       │   ├── newopen.cpython-36.pyc
│       │   ├── osCommands.cpython-36.pyc
│       │   ├── persepolis.cpython-36.pyc
│       │   ├── play.cpython-36.pyc
│       │   ├── progress.cpython-36.pyc
│       │   ├── properties.cpython-36.pyc
│       │   ├── setting.cpython-36.pyc
│       │   ├── shutdown.cpython-36.pyc
│       │   ├── spider.cpython-36.pyc
│       │   ├── startup.cpython-36.pyc
│       │   ├── text_queue.cpython-36.pyc
│       │   └── update.cpython-36.pyc
│       ├── setting.py
│       ├── shutdown.py
│       ├── spider.py
│       ├── startup.py
│       ├── text_queue.py
│       └── update.py
├── README.md
├── requirements.md
├── setup.py
├── test
│   └── test.py
├── uninstall.py
└── xdg
    └── persepolis.desktop

14 directories, 210 files
```
